### PR TITLE
Gui > Drawable and lots of connected changes

### PR DESCRIPTION
**Pull request by copygirl**
Apologies for the unhandy PR size. I was kind of sick not being able to map some thing because I wanted to focus on something and "do it the right (git) way". Perhaps I'll be able to split these changes off into multiple branches (and thus PRs) or at the very least multiple commits. Sorry~

Changes include:
- Renaming `Gui` to `Drawable`, `GuiScreen` to `Gui`
- HUD elements named `Hud...`, widgets `Widget...`
- Renamed some abstract (entity?) classes to `...Base`
- Named a couple containers (though most were named. changed some of them though)
- `Gui.update`and `Gui.draw` (used to be `drawScreen`)
- `ContainerChest` now `ContainerGeneric`
- `ScoreboardEntry`
- `RunArgs`
- `BlockRegion`
- `LocalDifficulty` (Only named class)
- Much more...

### CODE REVIEW

File: mappings/net/minecraft/client/gui/ingame/GuiStructureBlock.mapping [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
+CLASS none/bhx net/minecraft/client/gui/ingame/GuiStructureBlock
 	FIELD f LOGGER Lorg/apache/logging/log4j/Logger;
 	FIELD g te Lnone/arv;
 	METHOD <init> (Lnone/arv;)V
 		ARG 0 te
 	METHOD a handleKeyPress (CI)V
-	METHOD a drawScreen (IIF)V
+	METHOD a draw (IIF)V
 	METHOD a handleMouseClick (III)V
 	METHOD a handleButtonPressed (Lnone/bei;)V
```
Comment (asiekierka) [inline mappings/net/minecraft/client/gui/ingame/GuiStructureBlock.mapping:9]: General comment - if we have handleKeyPress, shouldn't this be handleButtonPress?

**Comment by asiekierka** [comment]
LGTM

**Comment by jamierocks** [comment]
I think the drawable package should merge with the gui package IMO

**Comment by asiekierka** [comment]
What about gui/ -> drawable/gui/?

**Comment by jamierocks** [comment]
I personally think that everything currently in drawable should be placed in gui

**Comment by asiekierka** [comment]
Sorry for not waiting longer, but my mapping work started clashing with copy's quite heavily, so I merged it to prevent further merge conflicts.

**Pull request merged**